piptools/utils.py

```diff
@@ -678,14 +678,34 @@ def parse_config_file(
             hint=f"Could not parse '{config_file !s}': {value_err !s}",
         )
 
-    # In a TOML file, we expect the config to be under `[tool.pip-tools]`
+    # In a TOML file, we expect the config to be under `[tool.pip-tools]`,
+    # `[tool.pip-compile]` or `[tool.pip-sync]`
     piptools_config: dict[str, Any] = config.get("tool", {}).get("pip-tools", {})
-    piptools_config = _normalize_keys_in_config(piptools_config)
-    piptools_config = _invert_negative_bool_options_in_config(
+    pipcompile_config: dict[str, Any] = (
+        config.get("tool", {}).get("pip-tools", {}).get("compile", {})
+    )
+    pipsync_config: dict[str, Any] = (
+        config.get("tool", {}).get("pip-tools", {}).get("sync", {})
+    )
```

Comment:
No need to double-lookup the dicts. The entire section is already in the var:
```suggestion
    pipcompile_config: dict[str, Any] = piptools_config.get("compile", {})
    pipsync_config: dict[str, Any] = piptools_config.get("sync", {})
```